### PR TITLE
Support proper destination format in weighted rule

### PR DESCRIPTION
**Pull request by rshriram**
**What this PR does / why we need it**:
Fixes the TODO (and regression) in route.go, where destination field in DestinationWeight was not honored.
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: proxy/envoy/route.go [hunk 1]
```diff
@@ -150,9 +150,11 @@ func buildHTTPRoute(config model.Config, service *model.Service, port *model.Por
 	if len(rule.Route) > 0 {
 		route.WeightedClusters = &WeightedCluster{}
 		for _, dst := range rule.Route {
-			// TODO: add destination service override from the route weights
-
-			cluster := buildOutboundCluster(destination, port, dst.Labels)
+			real_destination := destination
```
Comment (kyessenov) [inline proxy/envoy/route.go]: go convention is mixed caps (`actualDestination`)

File: model/conversion_test.go [hunk 2]
```diff
@@ -47,11 +47,15 @@ func TestProtoSchemaConversions(t *testing.T) {
 		"precedence": 5,
 		"route": [
 		{
-			"destination": "bar",
+			"destination": {
+                "name" : "bar"
```
Comment (rkpagadala) [inline model/conversion_test.go]: fix indent, here and below

Comment (rshriram) [reply model/conversion_test.go]: It looks fine in Gogland.. any idea why this is happening?

Comment (kyessenov) [reply model/conversion_test.go]: Maybe tabs?

Comment (kyessenov) [reply model/conversion_test.go]: Actually github messes up whitespace. It might be fine despite what web page shows.

Comment (ldemailly) [reply model/conversion_test.go]: you have tabs you shouldn't use tabs

**Comment by ldemailly** [comment]
it's odd this PR didn't pick up the breakage it is causing in consul
we should probably figure out why (off by one on linters vs workspace ?)
https://github.com/istio/pilot/pull/1275#issuecomment-329653627

**Comment by rshriram** [comment]
Determined that this is probably not the cause of breakage. @GregHanson thoughts?

**Comment by ldemailly** [comment]
yeah it's unrelated- seems to be a linter upgrade

**Pull request merged**